### PR TITLE
front: fix train on map (ts v2)

### LINKED ISSUE #7578: TSv2: Train on simulation map is gone
### What happened?

With trainschedule v2 activated, we don't have the train moving on the simulation map anymore.

It should be like that : 
Capture d’écran 2024-05-31 à 22 46 41

### What did you expect to happen?

To see the train and it should move when pressing play with the time button at the top of the screen.

### How can we reproduce it (as minimally and precisely as possible)?

1. Go to a scenario
2. Press play on the time button at the top of the simulation results and increase the multiplier
3. Check the simulation map at the bottom of the simulation results

### What operating system, browser and environment are you using?

- **Browser**: Firefox v126
- **OS**: macOS Sonoma 14.5
- **Env**: Dev

### OSRD version (top right corner `Account` button > `Informations`)

dev 4ae7056

### PR DESCRIPTION
**Pull request by clarani**
fix #7578

### CODE REVIEW

File: front/src/modules/simulationResult/components/SimulationResultsMapV2.tsx [hunk 1]
```diff
@@ -50,67 +45,58 @@ import { removeSearchItemMarkersOnMap } from 'common/Map/utils';
...
-  const simulation = useSelector(getPresentSimulation);
-  const trains = useMemo(() => keyBy(simulation.trains, 'id'), [simulation.trains]);
-  const selectedTrain = useSelector(getSelectedTrain);
+  const { isPlaying } = useSelector((state: RootState) => state.osrdsimulation);
   const terrain3DExaggeration = useSelector(getTerrain3DExaggeration);
   const layersSettings = useSelector(getLayersSettings);
 
+  const [mapLoaded, setMapLoaded] = useState(false);
+  const [interactiveLayerIds, setInteractiveLayerIds] = useState<string[]>([]);
+  const [selectedTrainHoverPosition, setTrainHoverPosition] = useState<TrainCurrentInfo>();
```
Comment (SharglutDev) [inline front/src/modules/simulationResult/components/SimulationResultsMapV2.tsx]:
```suggestion
  const [selectedTrainHoverPosition, setSelectedTrainHoverPosition] = useState<TrainCurrentInfo>();
```

**Review by SharglutDev — COMMENTED** [review]
Really nice refacto ! Not tested.

**Review by Uriel-Sautron — COMMENTED** [review]
Under review

**Review by Uriel-Sautron — APPROVED** [review]

File: front/src/modules/simulationResult/components/SimulationResultsMapV2.tsx (hunk 1 above)
Comment (clarani) [reply front/src/modules/simulationResult/components/SimulationResultsMapV2.tsx]: Done :)

File: front/src/applications/operationalStudies/views/v2/SimulationResultsV2.tsx [hunk 2]
```diff
@@ -240,14 +240,23 @@ const SimulationResultsV2 = ({
             <SimulationResultsMapV2
               setExtViewport={setExtViewport}
               geometry={pathProperties?.geometry}
+              trainSimulation={
+                selectedTrain && selectedTrainRollingStock && trainSimulation
```
Comment (kmer2016) [inline front/src/applications/operationalStudies/views/v2/SimulationResultsV2.tsx]: why do we need to check `selectedTrainRollingStock` here while is not used in the construction of the object below ?

File: front/src/common/Map/components/TrainOnMap/getTrainBody.ts [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+    upWidth: 0.019,
+    down: 0.02,
+  };
+  const tail = {
+    left: 0.05,
+    right: 0.05,
+    up: 0.05,
+    upWidth: 0.019,
+    down: 0.02,
+  };
```
Comment (kmer2016) [inline front/src/common/Map/components/TrainOnMap/getTrainBody.ts]:
```suggestion
  const tail = { ...head, up: 0.05 };
```

**Comment by anisometropie** [comment]
Simulation is going backward now
https://github.com/OpenRailAssociation/osrd/assets/24652545/28b7f95e-8100-4a18-9876-b18548a6f89a

File: front/src/applications/operationalStudies/views/v2/SimulationResultsV2.tsx (hunk 2 above)
Comment (clarani) [reply front/src/applications/operationalStudies/views/v2/SimulationResultsV2.tsx]: You're right, it's not necessary, I removed it :+1:

File: front/src/common/Map/components/TrainOnMap/getTrainBody.ts (hunk 3 above)
Comment (clarani) [reply front/src/common/Map/components/TrainOnMap/getTrainBody.ts]: Nice catch, thanks :+1:

**Review by SarahBellaha — APPROVED** [review]
LGTM, nice work 💪

**Review by kmer2016 — APPROVED** [review]
LGTM ! Good job !

**Pull request merged**